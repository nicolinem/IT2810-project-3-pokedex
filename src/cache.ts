import { gql, makeVar, useQuery } from '@apollo/client';
import {
  ApolloProvider,
  ApolloClient,
  HttpLink,
  ApolloLink,
  InMemoryCache,
} from '@apollo/client';


const httpLink = new HttpLink({
    uri: "http://it2810-01.idi.ntnu.no:4001/graphql",
});

// 7687

const authLink = new ApolloLink((operation, forward) => {
  // get the authentication token from local storage if it exists
  const token = localStorage.getItem('token');
  // return the headers to the context so httpLink can read them

  operation.setContext({
    headers: {
      authorization: token ? `Bearer ${token}` : '',
    },
  });

  // Call the next link in the middleware chain.
  return forward(operation);
});

export const isLoggedInVar = makeVar<boolean>(!!localStorage.getItem("token"));

export const cache: InMemoryCache = new InMemoryCache({
  typePolicies: {
    Query: {
      fields: {
        isLoggedIn: {
          read() {
            return isLoggedInVar();
          },
        },
      },
    },
  },
});

export const typeDefs = gql`
  extend type Query {
    isLoggedIn: Boolean!
  }
`;

const IS_LOGGED_IN = gql`
  query IsUserLoggedIn {
    isLoggedIn @client
  }
`;

export function IsLoggedIn() {
  const { data } = useQuery(IS_LOGGED_IN);
  return data.isLoggedIn;
}

export const client = new ApolloClient({
 link: authLink.concat(httpLink),
  cache: cache,
});
