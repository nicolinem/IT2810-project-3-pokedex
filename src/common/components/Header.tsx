import React from "react";
import { Accordion } from "./button/Accordion";
import SearchForm from "./SearchForm";

type Props = {
  children?: React.ReactNode;
  onClick?: () => void;
  className?: string;
  href?: string;
};
const Header: React.FC<Props> = ({ children, onClick }) => {
  return (
    <div className="bg-[#41444a] text-center h-96 ">
      <header className="text-4xl font-extrabold tracking-widest py-24">
        POKEDEX
      </header>
      <SearchForm></SearchForm>
      <div className="justify-items-center	">
        <Accordion
          title={"Advanced Search"}
          content={"fnejkfnejkn"}
        ></Accordion>
      </div>
    </div>
  );
};

export default Header;
