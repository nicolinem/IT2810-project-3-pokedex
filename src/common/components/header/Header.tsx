import React from "react";
import { Accordion } from "../button/Accordion";
import InputField from "../InputField";
import {SearchForm} from "../SearchForm";
import {TypeButtonContainer} from "../TypeButtonContainer";

type Props = {
  children?: React.ReactNode;
  onClick?: () => void;
  className?: string;
  href?: string;
};
const Header: React.FC<Props> = ({ children, onClick }) => {
  return (
    <div className="bg-[#121A36] text-center h-auto ">
      <header className="text-4xl font-extrabold text-white tracking-widest py-24">
        POKEDEX
      </header>
      <SearchForm></SearchForm>
    </div>
  );
};

export default Header;
