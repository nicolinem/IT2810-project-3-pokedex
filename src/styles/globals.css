

.TypeButton {
    position: absolute;
    width: 90px;
    height: 27px;
    left: 308px;
    bottom: 1188px;
    border-radius: 20px;
}