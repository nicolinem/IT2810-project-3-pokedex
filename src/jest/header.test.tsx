import { ApolloProvider } from "@apollo/client";
import { expect } from "@jest/globals";
import renderer from "react-test-renderer";
import { client } from "../cache";
import Header from "../common/components/header/Header";

describe("Header", () => {
  it("snapshot", () => {
    const tree = renderer
      .create(
        <ApolloProvider client={client}>
          <Header></Header>
        </ApolloProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
