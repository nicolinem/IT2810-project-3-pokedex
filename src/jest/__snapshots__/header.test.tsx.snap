// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot 1`] = `
<div
  className="bg-[#121A36] text-center h-44 "
>
  <header
    className="text-4xl font-extrabold text-white tracking-widest py-32"
  >
    P o k Ã© d e x
  </header>
  <div
    className="absolute right-5 top-5 z-40"
  >
    <button
      className="text-white text-sm rounded-full p-2 px-4 hover:bg-gray-600 active:bg-gray-700 focus:outline-none flex-row bg-gray-500 undefined"
      onClick={[Function]}
    >
       Sign in
    </button>
  </div>
</div>
`;
